gamemodes/terrortown/gamemode/client/cl_help_data.lua

```diff
@@ -0,0 +1,254 @@
+-- @realm client
+function HELP_DATA_OBJECT:AdminOnly(adminOnly)
+	self.adminOnly = adminOnly == nil and true or adminOnly
+end
+
+---
+-- Sets a callback function that can be used to descide at a
+-- later point if a menu should be shown
+-- @param function fn The callback function
+-- @realm client
+function HELP_DATA_OBJECT:ShouldShow(fn)
+	self.shouldShowFn = fn
+end
+
+---
+-- Sets a callback function that is called when the menu button
+-- is clicked
+-- @param function fn The callback function
+-- @realm client
+function HELP_DATA_OBJECT:OnClick(fn)
+	self.onClickFn = fn
+end
+
+
+SUB_HELP_DATA = {}
+
+local SUB_HELP_DATA_OBJECT = {}
+
+---
+-- Binds data table to the @{SUB_HELP_DATA} object
+-- @param table data The data table with all navigation points
+-- @return @{SUB_HELP_DATA} The object to be used in the hook
+-- @internal
+-- @realm client
+function SUB_HELP_DATA:BindData(menuTbl)
+	self.menuTbl = menuTbl or {}
+
+	return self
+end
+
+---
+-- Populates a given submenu with navigation points
+-- @param string id The unique ID of the navigation point
+-- @return table A reference to the new table
+-- @realm client
+function SUB_HELP_DATA:PopulateSubMenu(id)
+	if self:Exists(id) then return end
+
+	local pos = #self.menuTbl + 1
+
+	self.menuTbl[pos] = table.Copy(SUB_HELP_DATA_OBJECT)
+	self.menuTbl[pos].id = id
+
+	return self.menuTbl[pos]
+end
+
+---
+-- Checks if a submenu with the given ID is already registered
+-- @param string id The unique submenu identifier
+-- @return boolean Returns if the identifier is already used
+-- @realm client
+function SUB_HELP_DATA:Exists(id)
+	for i = 1, #self.menuTbl do
+		if self.menuTbl[i].id == id then
+			return true
+		end
+	end
+
+	return false
+end
+
+---
+-- Sets the title of a submenu element
+-- @param string title The name, can be a language identifier
+-- @realm client
+function SUB_HELP_DATA_OBJECT:SetTitle(title)
+	self.title = title
+end
+
+---
+-- Sets the admin only state of a submenu element
+-- @param [default=true] boolean adminOnly Set true to show this menu only to admins
+-- @realm client
+function SUB_HELP_DATA_OBJECT:AdminOnly(adminOnly)
+	self.adminOnly = adminOnly == nil and true or adminOnly
+end
+
+---
+-- Callback function that is used to call code to populate
+-- the main panel of a submenu
+-- @param function fn The callback function
+-- @realm client
+function SUB_HELP_DATA_OBJECT:PopulatePanel(fn)
+	self.populateFn = fn
+end
+
+---
+-- Callback function that is used to call code to populate
+-- the button panel of a submenu, this panel is only created if
+-- this callback function is set
+-- @param function fn The callback function
+-- @realm client
+function SUB_HELP_DATA_OBJECT:PopulateButtonPanel(fn)
+	self.populateButtonFn = fn
+end
+
+---
+-- Sets a callback function that can be used to descide at a
+-- later point if a submenu should be shown
+-- @param function fn The callback function
+-- @realm client
+function SUB_HELP_DATA_OBJECT:ShouldShow(fn)
```

Comment:
Rename the function to indicate, that this will add a callback and not answer the question if this should be shown